lib/house.rb

```diff
@@ -0,0 +1,69 @@
+class House
+  attr_reader :price,
+              :address,
+              :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    @price.gsub('$','').to_i > 500000
+  end
+
+  def rooms_from_category(x)
+    @rooms.select do |room|
+      if room.category == x
+      room
+      end
+    end
+  end
+
+  def area
+    house_room_area_array.inject{ |sum,x| sum + x }
+  end
+
+  def house_room_area_array
```

Comment:
I don't think this works since we want to return `total_area` from this method. You might want to use `#inject` like you are using above to calculate total_area